server/app/models/fusor/deployment.rb

```diff
@@ -29,15 +29,21 @@ class Deployment < ActiveRecord::Base
     belongs_to :rhev_engine_host, :class_name => "::Host::Base", :foreign_key => :rhev_engine_host_id
     # if we want to envorce discovered host uniqueness uncomment this line
     #validates :rhev_engine_host_id, uniqueness: { :message => _('This Host is already a RHEV Engine for a different deployment') }
-    has_many :rhev_hypervisor_hosts, :class_name => "::Host::Base", :through => :deployment_hosts, :source => :discovered_host
+    has_many :rhev_hypervisor_hosts, :class_name => "::Host::Base", :through => :deployment_hypervisor_hosts, :source => :discovered_host
     validates_with ::Katello::Validators::KatelloNameFormatValidator, :attributes => :name
-
+ 
     belongs_to :discovered_host, :class_name => "::Host::Base", :foreign_key => :rhev_engine_host_id
-    has_many :deployment_hosts, :class_name => "Fusor::DeploymentHost", :foreign_key => :deployment_id, :inverse_of => :deployment
-    has_many :discovered_hosts, :through => :deployment_hosts, :foreign_key => :discovered_host_id, :source => :discovered_host
+    
+    has_many :deployment_hypervisor_hosts, :class_name => "Fusor::DeploymentHost", :conditions => {:deployment_host_type => 'rhev_hypervisor' }
+    has_many :discovered_hosts, :through => :deployment_hypervisor_hosts, :class_name => "::Host::Base", :source => :discovered_host
```

Comment:
Add a TODO comment above discovered_hosts stating we need to rename that to hypervisor_hosts.